includes/sanitizers/class-amp-meta-sanitizer.php

```diff
@@ -148,6 +150,37 @@ protected function ensure_viewport_is_present() {
 		}
 	}
 
+	/**
+	 * Always ensure we have a <style amp-boilerplate> tag.
+	 */
+	protected function ensure_boilerplate_is_present() {
+		$style = $this->dom->xpath->query( './style[ @amp-boilerplate ]', $this->dom->head )->item( 0 );
+
+		if ( ! $style ) {
+			$style = $this->dom->createElement( Tag::STYLE );
+			$style->setAttribute( Attribute::AMP_BOILERPLATE, '' );
+			$style->appendChild( $this->dom->createTextNode( amp_get_boilerplate_stylesheets()[0] ) );
+		} else {
+			$style->parentNode->removeChild( $style ); // So we can move it.
+		}
+
+		$this->dom->head->appendChild( $style );
+
+		$noscript = $this->dom->xpath->query( './noscript[ style[ @amp-boilerplate ] ]', $this->dom->head )->item( 0 );
+
+		if ( ! $noscript ) {
+			$noscript = $this->dom->createElement( Tag::NOSCRIPT );
+			$style    = $this->dom->createElement( Tag::STYLE );
+			$style->setAttribute( Attribute::AMP_BOILERPLATE, '' );
+			$style->appendChild( $this->dom->createTextNode( amp_get_boilerplate_stylesheets()[1] ) );
+			$noscript->appendChild( $style );
+		} else {
+			$noscript->parentNode->removeChild( $noscript ); // So we can move it.
+		}
+
+		$this->dom->head->appendChild( $noscript );
+	}
```

Comment:
This doesn't really make sense to be part of the Meta sanitizer, if “Meta” is referring to `meta` tags, but we can move it into a separate sanitizer later.

Reply:
Or rather, maybe this sanitizer should be renamed to be something like `AMP_Ensure_Required_Markup_Sanitizer`.